### PR TITLE
[fix] crash on some android devices

### LINKED ISSUE #7965: Improve makefile to only make Nix install the bare minimum for intended platform
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

### Summary
[comment]: # (Summarise the problem and how the pull request solves it)
This PR allows the Makefile to leverage the target-os argument to only install the packages that are required for what we're building. E.g. a new user that clones the repo on iOS and runs `make release-ios` won't see the Android SDK, NDK, gradle, etc being downloaded.

status: ready

### LINKED ISSUE #8366: Can't launch mobile app after realm upgrade: Unfortunately, Status has stopped (Android 6.0.1, Galaxy S6 and S7)
### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Bug

[comment]: # (Describe the feature you would like, or briefly summarise the bug and what you did, what you expected to happen, and what actually happens. Sections below)
*Summary*: Can't launch app, see "Unfortunately, Status has stopped" popup instead. Installed Android nightly build from https://ci.status.im/job/status-react/job/nightly/1105/

Note that previous nightly https://ci.status.im/job/status-react/job/nightly/1104/ can be installed and launched fine. Assume that the issues is introduced by https://github.com/status-im/status-react/pull/8336
#### Expected behavior

[comment]: # (Describe what you expected to happen.)
app launched
#### Actual behavior
 "Unfortunately, Status has stopped" popup 

logcat contains
```
06-05 15:49:48.421  3059  3059 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
06-05 15:49:48.421  3059  3059 F DEBUG   : Build fingerprint: 'samsung/zerofltexx/zeroflte:6.0.1/MMB29K/G920FXXU5DQA8:user/release-keys'
06-05 15:49:48.421  3059  3059 F DEBUG   : Revision: '11'
06-05 15:49:48.421  3059  3059 F DEBUG   : ABI: 'arm'
06-05 15:49:48.421  3059  3059 F DEBUG   : pid: 8122, tid: 8172, name: mqt_js  >>> im.status.ethereum <<<
06-05 15:49:48.421  3059  3059 F DEBUG   : signal 11 (SIGSEGV), code 2 (SEGV_ACCERR), fault addr 0x460eb084
06-05 15:49:48.421  3507  3507 I KnoxTimeoutHandler: SD activityfalse
06-05 15:49:48.421  3050  3050 D libEGL  : eglTerminate EGLDisplay = 0x7ffb963b98
06-05 15:49:48.421  3059  3059 F DEBUG   :     r0 00000000  r1 00000000  r2 f5110ee1  r3 460eb084
06-05 15:49:48.421  3059  3059 F DEBUG   :     r4 d82a2b88  r5 00000000  r6 d406f3ac  r7 d406f300
06-05 15:49:48.421  3059  3059 F DEBUG   :     r8 eec6fa1c  r9 eed16470  sl eec6fa14  fp f73f9ec0
06-05 15:49:48.421  3059  3059 F DEBUG   :     ip d82a2d14  sp d406f2c0  lr d7da8538  pc 460eb084  cpsr 600e0010
06-05 15:49:48.431  4435  4435 D Launcher: onTrimMemory. Level: 20
06-05 15:49:48.431  3059  3059 F DEBUG   : 
06-05 15:49:48.431  3059  3059 F DEBUG   : backtrace:
06-05 15:49:48.431  3059  3059 F DEBUG   :     #00 pc 134ea084  /dev/ashmem/dalvik-main space 1 (deleted) (offset 0x1000)
06-05 15:49:48.431  3059  3059 F DEBUG   :     #01 pc 0021b534  /data/app/im.status.ethereum-1/lib/arm/librealmreact.so
06-05 15:49:48.431  3059  3059 F DEBUG   :     #02 pc 0002063d  /data/app/im.status.ethereum-1/lib/arm/libjscexecutor.so (_ZN8facebook3jsc10JSCRuntimeC1Ev+16)
06-05 15:49:48.431  3059  3059 F DEBUG   :     #03 pc 0002264b  /data/app/im.status.ethereum-1/lib/arm/libjscexecutor.so (_ZN8facebook3jsc14makeJSCRuntimeEv+14)
06-05 15:49:48.431  3059  3059 F DEBUG   :     #04 pc 00018c0f  /data/app/im.status.ethereum-1/lib/arm/libjscexecutor.so
06-05 15:49:48.431  3059  3059 F DEBUG   :     #05 pc 000a56f3  /data/app/im.status.ethereum-1/lib/arm/libreactnativejni.so (_ZN8facebook5react16NativeToJsBridgeC1EPNS0_17JSExecutorFactoryENSt6__ndk110shared_ptrINS0_14ModuleRegistryEEENS5_INS0_18MessageQueueThreadEEENS5_INS0_16InstanceCallbackEEE+146)
06-05 15:49:48.431  3059  3059 F DEBUG   :     #06 pc 000a0913  /data/app/im.status.ethereum-1/lib/arm/libreactnativejni.so
06-05 15:49:48.431  3059  3059 F DEBUG   :     #07 pc 000867df  /data/app/im.status.ethereum-1/lib/arm/libreactnativejni.so
06-05 15:49:48.431  3059  3059 F DEBUG   :     #08 pc 0008531b  /data/app/im.status.ethereum-1/lib/arm/libreactnativejni.so
06-05 15:49:48.431  3059  3059 F DEBUG   :     #09 pc 000791af  /data/app/im.status.ethereum-1/lib/arm/libreactnativejni.so (_ZN8facebook3jni6detail13MethodWrapperIMNS_5react15JNativeRunnableEFvvEXadL_ZNS4_3runEvEES4_vJEE8dispatchENS0_9alias_refIPNS1_8JTypeForINS0_11HybridClassIS4_NS3_8RunnableEE8JavaPartESB_vE11_javaobjectEEE+10)
06-05 15:49:48.431  3059  3059 F DEBUG   :     #10 pc 000791e9  /data/app/im.status.ethereum-1/lib/arm/libreactnativejni.so (_ZN8facebook3jni6detail17WrapForVoidReturnIPFvNS0_9alias_refIPNS1_8JTypeForINS0_11HybridClassINS_5react15JNativeRunnableENS6_8RunnableEE8JavaPartES8_vE11_javaobjectEEEEXadL_ZNS1_13MethodWrapperIMS7_FvvEXadL_ZNS7_3runEvEES7_vJEE8dispatchESE_EEvJSD_EE4callEOSD_+20)
06-05 15:49:48.431  3059  3059 F DEBUG   :     #11 pc 0007915f  /data/app/im.status.ethereum-1/lib/arm/libreactnativejni.so (_ZN8facebook3jni6detail15FunctionWrapperIPFvNS0_9alias_refIPNS1_8JTypeForINS0_11HybridClassINS_5react15JNativeRunnableENS6_8RunnableEE8JavaPartES8_vE11_javaobjectEEEEXadL_ZNS1_13MethodWrapperIMS7_FvvEXadL_ZNS7_3runEvEES7_vJEE8dispatchESE_EESD_vJEE4callEP7_JNIEnvP8_jobject+30)
06-05 15:49:48.431  3059  3059 F DEBUG   :     #12 pc 015ef9b7  /data/app/im.status.ethereum-1/oat/arm/base.odex (offset 0x13e4000)
```

[comment]: # (Describe what actually happened.)

### Reproduction

[comment]: # (Describe how we can replicate the bug step by step.)

- Install Status
- Launch app

### Additional Information

[comment]: # (Please do your best to fill this out.)
- Status version: nightly 
[comment]: # (Android or iOS.)
- Operating System: Android 6.0.1 real Galaxy S6 and real Galaxy S7 from browserstack with Android 6

#### Logs

[comment]: # (if on Android please replicate bug whilst running adb logcat)

full logcat: 
[cantstart.txt](https://github.com/status-im/status-react/files/3257552/cantstart.txt)

### PR DESCRIPTION
**Pull request by yenda**
fix #8366 @annadanchenko that might fix our problem

Steps:

Install app
Launch app. Do not create any account
kill app
Launch app again

Can reproduce issue

Galaxy S6, Android 6.0.1
Galaxy S7, Android 6
Galaxy S7, Android 8.0.0
Galaxy S8, Android 7
Can't reproduce issue:

Galaxy Note 4, Android 6.0.1
Nexus 6, Android 6

status: ready

### CODE REVIEW

**Comment by jakubgs** [comment]
Oh, so this is an alternative possible fix for #8366?

The release notes for `2.28.1` say:
>Fixed
>A bug in instruction cache invalidation could lead to SIGBUS errors on RN Android. (#2391, since v2.28.0)
https://github.com/realm/realm-js/releases/tag/v2.28.1

It's true, https://github.com/realm/realm-js/issues/2391 does look related.

**Review by jakubgs — APPROVED** [review]

**Comment by yenda** [comment]
@annadanchenko can you try on one of the affected devices?

File: mobile_files/package.json.orig [hunk 1]
```diff
@@ -65,7 +65,7 @@
     "react-native-webview": "^5.2.1",
     "react-native-webview-bridge": "git+https://github.com/status-im/react-native-webview-bridge.git#fix/classnames-colision",
     "react-navigation": "^2.12.1",
-    "realm": "2.28.0",
+    "realm": "^2.28.1",
```
Comment (pedropombeiro) [inline mobile_files/package.json.orig]: I think we're tending towards using specific version numbers (and upgrade explicitly as necessary). At least that's what I've been seeing @yenda doing.

**Review by pedropombeiro — APPROVED** [review]
Approved with a question

File: mobile_files/package.json.orig (hunk 1 above)
Comment (jakubgs) [reply mobile_files/package.json.orig]: Ah, you're right, I didn't notice the `^`. @yenda can you please drop it?

Comment (yenda) [reply mobile_files/package.json.orig]: yes I will I just want a quick build to check if it fixes the buck, it's stupid yarn that did that, is there a way to make it upgrade a lib without acting out like this?

Comment (jakubgs) [reply mobile_files/package.json.orig]: `yarn upgrade --exact`?

Comment (jakubgs) [reply mobile_files/package.json.orig]: It's still using the `^`.

**Comment by churik** [comment]
@yenda android builds are failed with error (tried to relaunch several times, didn't help):
```
2019-06-06T15:22:04.484Z] FAILURE: Build failed with an exception.

[2019-06-06T15:22:04.484Z] 

[2019-06-06T15:22:04.484Z] * What went wrong:

[2019-06-06T15:22:04.484Z] Execution failed for task ':app:bundlePrJsAndAssets'.

[2019-06-06T15:22:04.484Z] > Process 'command 'node'' finished with non-zero exit value 1

[2019-06-06T15:22:04.484Z] 
```
Can you take a look?

**Comment by yenda** [comment]
@churik please use the first one

**Comment by annadanchenko** [comment]
@yenda @jakubgs @churik crash can't be reproduced anymore on build 1 for android. Me and @churik will do a bit more testing for the next steps after launch ...

**Comment by yenda** [comment]
@annadanchenko thanks I will fill a proper PR then

**Comment by churik** [comment]
@yenda can confirm that it is not reproducible on build 1 with given steps in #8366 and steps in description of this PR (tried on Android 6, Galaxy S7 BrowserStack device)
So what's the plan now - to check possible regression here or wait for another PR?
Not clear to me what is the meaning of https://github.com/status-im/status-react/pull/8368#issuecomment-499801014

**Comment by yenda** [comment]
@churik this was a quick pr to check if the realm bugfix was working, so it is just a version minor version bump of realm. There is nothing to check besides the end of crashes on these specific devices. 
However as you can see there is some builds failing, that is what I mean by filling a proper PR.

**Comment by yenda** [comment]
@churik done

**Comment by jakubgs** [comment]
So, are we merging this or what?

**Comment by yenda** [comment]
just waiting for to merge flag

**Comment by churik** [comment]
Rechecked that can install and create an account on Galaxy S7 (Android 6), Google Nexus 6(Android 6), IPhone7 (IOS 11, just in case).
Also checked basically fetching messages from offline, scan QR feature, relogin, enabling preview mode and biometric authentications.
Looks good to me.

**Review by jakubgs — APPROVED** [review]
Snitches get stitches bot...

**Pull request merged**